src/Nethermind/Nethermind.Core/Encoding/RlpStream.cs

```diff
@@ -509,44 +525,59 @@ public int ReadSequenceLength()
         private int DeserializeLength(int lengthOfLength)
         {
             int result;
-            if (Data[Position] == 0)
+            if (PeekByte() == 0)
             {
                 throw new RlpException("Length starts with 0");
             }
 
             if (lengthOfLength == 1)
             {
-                result = Data[Position];
+                result = PeekByte();
             }
             else if (lengthOfLength == 2)
             {
-                result = Data[Position + 1] | (Data[Position] << 8);
+                result = PeekByte(1) | (PeekByte() << 8);
             }
             else if (lengthOfLength == 3)
             {
-                result = Data[Position + 2] | (Data[Position + 1] << 8) | (Data[Position] << 16);
+                result = PeekByte(2) | (PeekByte(1) << 8) | (PeekByte()<< 16);
             }
             else if (lengthOfLength == 4)
             {
-                result = Data[Position + 3] | (Data[Position + 2] << 8) | (Data[Position + 1] << 16) |
-                         (Data[Position] << 24);
+                result = PeekByte(3) | (PeekByte(2) << 8) | (PeekByte(1) << 16) |
+                         (PeekByte() << 24);
             }
             else
             {
                 // strange but needed to pass tests - seems that spec gives int64 length and tests int32 length
                 throw new InvalidOperationException($"Invalid length of length = {lengthOfLength}");
             }
 
-            Position += lengthOfLength;
+            SkipBytes(lengthOfLength);
             return result;
         }
 
-        public byte ReadByte()
+        public virtual byte ReadByte()
         {
             return Data[Position++];
         }
+        
+        protected virtual byte PeekByte()
```

Comment:
Do we need 2 PeekByte methods instead of one with parameter with default value (0 or null)?

Reply:
hmmm I see, good point, I have added the other one later